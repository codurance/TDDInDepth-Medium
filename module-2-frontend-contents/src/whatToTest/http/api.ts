import * as axios from "axios";
import {NewProduct} from "../types/NewProduct.ts";
import {Product} from "../types/Product.ts";
import {generateId} from "./generateId.ts";


const request = axios.default.create({
    baseURL: "http://localhost:5000",
})

export const createProductRequest = async (product: NewProduct): Promise<Product> => {
    const {data} = await request.post<Product>("/products", {
        id: generateId(),
        ...product,
    });

    return data;
}

export const getProductsRequest = async (): Promise<Product[]> => {
    const {data} = await request.get<Product[]>("/products");

    return data;
}