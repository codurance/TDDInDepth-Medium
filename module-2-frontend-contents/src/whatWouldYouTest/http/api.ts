import * as axios from "axios";
import {Todo} from "../types/Todo.ts";
import {NewTodo} from "../types/NewTodo.ts";
import {generateId} from "./generateId.ts";

const request = axios.default.create({
    baseURL: "http://localhost:5000",
})

export const getTodosRequest = async () => {
    const {data} = await request.get<Todo[]>("/todos");

    return data;
}

export const updateTodoRequest = async (todo: Todo) => {
    await request.patch(`/todos/${todo.id}`, todo);
}

export const createTodoRequest = async (newTodo: NewTodo) => {
    const todo: Todo = {id: generateId(), status: "todo", ...newTodo,};
    const {data} = await request.post<Todo>("/todos", todo);

    return data;
}