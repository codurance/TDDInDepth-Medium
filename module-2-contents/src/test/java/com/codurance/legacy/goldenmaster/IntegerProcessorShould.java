package com.codurance.legacy.goldenmaster;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import java.util.Arrays;
import java.util.List;

public class IntegerProcessorShould {
    @Test
    public void processIntegerNumbers() {
        IntegerProcessor processor = new IntegerProcessor();

        // Test inputs that cover various scenarios
        List<List<Integer>> testCases = Arrays.asList(
            Arrays.asList(1, 2, 3),                    // Simple positive numbers
            Arrays.asList(-1, -2, -3),                 // Simple negative numbers
            Arrays.asList(0, 0, 0),                    // All zeros
            Arrays.asList(101, 200),                   // Numbers greater than 100
            Arrays.asList(-101, -200),                 // Numbers less than -100
            Arrays.asList(50, -50, 0, 150, -150),      // Mixed numbers
            Arrays.asList()                            // Empty list
        );

        // Capture the output for each input
        StringBuilder results = new StringBuilder();
        for (List<Integer> testCase : testCases) {
            results.append("Input: ").append(testCase).append("\n");
            results.append("Output: ").append(processor.process(testCase)).append("\n\n");
        }

        // Verify with Approvals
        Approvals.verify(results.toString());
    }
}
