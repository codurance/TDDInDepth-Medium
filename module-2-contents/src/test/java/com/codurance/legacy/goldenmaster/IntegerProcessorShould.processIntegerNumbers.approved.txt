Input: [1, 2, 3]
Output: 6

Input: [-1, -2, -3]
Output: 6

Input: [0, 0, 0]
Output: 0

Input: [101, 200]
Output: 602

Input: [-101, -200]
Output: 150

Input: [50, -50, 0, 150, -150]
Output: 475

Input: []
Output: 0

